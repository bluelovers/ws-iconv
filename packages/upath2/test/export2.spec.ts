import _m0 = require('../');
import _m1 from '../';
import path from 'path';

test(`win32`, () =>
{

	let actual = _m0.win32;
	let expected = _m1.win32;

	expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();
	expect(Object.keys(actual)).toMatchSnapshot();

});

test(`posix`, () =>
{

	let actual = _m0.posix;
	let expected = _m1.posix;

	expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();
	expect(Object.keys(actual)).toMatchSnapshot();

});

describe(`path`, () => {

	Object.entries(path)
		.forEach(([name, prop]) => {

			const type = typeof prop;

			test(`${name} => ${type}`, () =>
			{
				let actual = _m0[name];
				let expected = _m1[name];

				expect(actual).toStrictEqual(expected);

				expect(typeof actual).toStrictEqual(type);
			})

		})
	;

})

describe(`hasOwnProperty`, () => {

	[
		'win32',
		'posix',
		'upath',
		'default',
	].forEach(key => {
		test(key, () =>
		{

			let actual = _m0.hasOwnProperty(key);
			let expected = _m1.hasOwnProperty(key);

			expect(actual).toStrictEqual(expected);
			expect(_m0[key]).toStrictEqual(_m1[key]);

			expect(actual).toBeTruthy();
			expect(actual).toMatchSnapshot();

		});
	})

})
