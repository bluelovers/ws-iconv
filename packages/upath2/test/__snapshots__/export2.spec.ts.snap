// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasOwnProperty default 1`] = `true`;

exports[`hasOwnProperty posix 1`] = `true`;

exports[`hasOwnProperty upath 1`] = `true`;

exports[`hasOwnProperty win32 1`] = `true`;

exports[`posix 1`] = `
PathWrap {
  "default": PathWrap {
    "PathWrap": [Function],
    "default": [Circular],
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "upath",
    "posix": [Circular],
    "sep": "/",
    "upath": [Circular],
    "win32": PathWrap {
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "win32",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
  },
  "fn": PathWrap {
    "_makeLong": [Function],
    "delimiter": ";",
    "filterNameEntry": [Function],
    "fn": PathWrap {
      "basename": [Function],
      "constructor": [Function],
      "dirname": [Function],
      "extname": [Function],
      "fn": [Circular],
      "format": [Function],
      "isAbsolute": [Function],
      "join": [Function],
      "normalize": [Function],
      "parse": [Function],
      "relative": [Function],
      "resolve": [Function],
      "sep": "/",
    },
    "sep": "/",
    "toNamespacedPath": [Function],
    "vaildNameEntry": [Function],
  },
  "name": "posix",
  "posix": [Circular],
  "sep": "/",
  "upath": PathWrap {
    "PathWrap": [Function],
    "default": [Circular],
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "upath",
    "posix": [Circular],
    "sep": "/",
    "upath": [Circular],
    "win32": PathWrap {
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "win32",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
  },
  "win32": PathWrap {
    "default": PathWrap {
      "PathWrap": [Function],
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "upath",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "win32",
    "posix": [Circular],
    "sep": "/",
    "upath": PathWrap {
      "PathWrap": [Function],
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "upath",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "win32": [Circular],
  },
}
`;

exports[`posix 2`] = `
Array [
  "sep",
  "name",
  "fn",
  "win32",
  "posix",
  "upath",
  "default",
]
`;

exports[`win32 1`] = `
PathWrap {
  "default": PathWrap {
    "PathWrap": [Function],
    "default": [Circular],
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "upath",
    "posix": PathWrap {
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "posix",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "sep": "/",
    "upath": [Circular],
    "win32": [Circular],
  },
  "fn": PathWrap {
    "_makeLong": [Function],
    "delimiter": ";",
    "filterNameEntry": [Function],
    "fn": PathWrap {
      "basename": [Function],
      "constructor": [Function],
      "dirname": [Function],
      "extname": [Function],
      "fn": [Circular],
      "format": [Function],
      "isAbsolute": [Function],
      "join": [Function],
      "normalize": [Function],
      "parse": [Function],
      "relative": [Function],
      "resolve": [Function],
      "sep": "/",
    },
    "sep": "/",
    "toNamespacedPath": [Function],
    "vaildNameEntry": [Function],
  },
  "name": "win32",
  "posix": PathWrap {
    "default": PathWrap {
      "PathWrap": [Function],
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "upath",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "posix",
    "posix": [Circular],
    "sep": "/",
    "upath": PathWrap {
      "PathWrap": [Function],
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "upath",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "win32": [Circular],
  },
  "sep": "/",
  "upath": PathWrap {
    "PathWrap": [Function],
    "default": [Circular],
    "fn": PathWrap {
      "_makeLong": [Function],
      "delimiter": ";",
      "filterNameEntry": [Function],
      "fn": PathWrap {
        "basename": [Function],
        "constructor": [Function],
        "dirname": [Function],
        "extname": [Function],
        "fn": [Circular],
        "format": [Function],
        "isAbsolute": [Function],
        "join": [Function],
        "normalize": [Function],
        "parse": [Function],
        "relative": [Function],
        "resolve": [Function],
        "sep": "/",
      },
      "sep": "/",
      "toNamespacedPath": [Function],
      "vaildNameEntry": [Function],
    },
    "name": "upath",
    "posix": PathWrap {
      "default": [Circular],
      "fn": PathWrap {
        "_makeLong": [Function],
        "delimiter": ";",
        "filterNameEntry": [Function],
        "fn": PathWrap {
          "basename": [Function],
          "constructor": [Function],
          "dirname": [Function],
          "extname": [Function],
          "fn": [Circular],
          "format": [Function],
          "isAbsolute": [Function],
          "join": [Function],
          "normalize": [Function],
          "parse": [Function],
          "relative": [Function],
          "resolve": [Function],
          "sep": "/",
        },
        "sep": "/",
        "toNamespacedPath": [Function],
        "vaildNameEntry": [Function],
      },
      "name": "posix",
      "posix": [Circular],
      "sep": "/",
      "upath": [Circular],
      "win32": [Circular],
    },
    "sep": "/",
    "upath": [Circular],
    "win32": [Circular],
  },
  "win32": [Circular],
}
`;

exports[`win32 2`] = `
Array [
  "sep",
  "name",
  "fn",
  "win32",
  "posix",
  "upath",
  "default",
]
`;
